### PR TITLE
RMIO-67: Upgrade to NUnit3

### PR DESCRIPTION
**Pull request by fillip-frackiewicz**
https://re-motion.atlassian.net/browse/RMIO-67

### CODE REVIEW

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class ZipFileBuilderTest
     private class NotClosableMemeoryStream : MemoryStream
     {
       public NotClosableMemeoryStream ([NotNull] byte[] buffer)
-        : base (buffer)
+          : base (buffer)
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:38]: Unnecessary whitespace change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 2]
```diff
@@ -152,7 +152,7 @@ public void BuildReturnsZipFileWithEmptyFile_FromInMemoryFile ()
     [Test]
     public void BuildReturnsZipFileWithFileWithUmlaut ()
     {
-      string fileWithUmlautInName = Path.Combine (Path.GetTempPath(), "NameWith�.txt");
+      string fileWithUmlautInName = Path.Combine (Path.GetTempPath(), "NameWith�.txt");
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:155]: Unnecessary encoding change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 3]
```diff
@@ -229,8 +235,8 @@ public void Build_IOExceptionDuringCopyingOccurs_ThrowsAbortExceptionWithIOExcep
               }
             },
             Throws.InstanceOf<AbortException>()
-                .With.Message.EqualTo (@"Error while copying the data from the file 'C:\fileName' to the archive.")
-                .And.InnerException.SameAs (ioException));
+                  .With.Message.EqualTo (@"Error while copying the data from the file 'C:\fileName' to the archive.")
+                  .And.InnerException.SameAs (ioException));
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:239]: Unnecessary whitespace changes

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 4]
```diff
@@ -319,6 +329,7 @@ public void SetFileProcessingAction_Retry ()
       using (zipBuilder.Build (zipFileName))
       {
       }
+
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:332]: Empty line

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 5]
```diff
@@ -348,7 +359,7 @@ public void BuildReturnsZipFilesWithFoldersAndFiles ()
       //--file2
       //--file3
       //-Directory2
-      //--Directory3 �
+      //--Directory3 �
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:362]: Encoding

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 6]
```diff
@@ -375,7 +386,7 @@ public void BuildReturnsZipFilesWithFoldersAndFiles ()
 
       var directory1 = Directory.CreateDirectory (Path.Combine (rootPath, "Directory1"));
       var directory2 = Directory.CreateDirectory (Path.Combine (rootPath, "Directory2"));
-      var directory3 = Directory.CreateDirectory (Path.Combine (directory2.FullName, "Directory3 �"));
+      var directory3 = Directory.CreateDirectory (Path.Combine (directory2.FullName, "Directory3 �"));
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:389]: Encoding

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 7]
```diff
@@ -401,24 +412,25 @@ public void BuildReturnsZipFilesWithFoldersAndFiles ()
       using (zipBuilder.Build (zipFileName))
       {
       }
+
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:415]: Empty line

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 8]
```diff
@@ -401,24 +412,25 @@ public void BuildReturnsZipFilesWithFoldersAndFiles ()
...
-                              Path.GetFileName (file3.FileName),
-                              Path.GetFileName (file4.FileName),
-                              Path.GetFileName (file5.FileName),
-                              Path.GetFileName (file6.FileName)
+                            Path.GetFileName (file1.FileName),
+                            Path.GetFileName (file2.FileName),
+                            Path.GetFileName (file3.FileName),
+                            Path.GetFileName (file4.FileName),
+                            Path.GetFileName (file5.FileName),
+                            Path.GetFileName (file6.FileName)
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:423]: Unnecessary whitespace changes

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 9]
```diff
@@ -401,24 +412,25 @@ public void BuildReturnsZipFilesWithFoldersAndFiles ()
...
-                                      file3NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
-                                      file4NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
-                                      file5NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
-                                      file6NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/")
+                                    file1NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
+                                    file2NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
+                                    file3NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
+                                    file4NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
+                                    file5NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/"),
+                                    file6NewLocation.Substring (commonRoot.FullName.Length + 1).Replace ("\\", "/")
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:433]: Unnecessary whitespace changes

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 10]
```diff
@@ -442,29 +453,39 @@ public void BuildThrowsAbortExceptionUponCancel ()
...
                       CreateFile ("file1", 50),
                       CreateFile ("file2", 60),
-                      CreateDirectory ("dir2",
+                      CreateDirectory (
+                          "dir2",
                           CreateFile ("file1", 70),
                           CreateFile ("file2", 80)))),
-              CreateDirectory ("dir3",
+              CreateDirectory (
+                  "dir3",
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:488]: These all seem like unnecessary whitspace changes to me

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 11]
```diff
@@ -514,7 +535,8 @@ public void BuildReportsProperly ()
     public void Build_FileSizeIsGreaterThanBufferSize_ReportsProperly ()
     {
       var root =
-          CreateDirectory ("root",
+          CreateDirectory (
+              "root",
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:539]: Unnecessary whitespace change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 12]
```diff
@@ -575,15 +597,17 @@ public void Build_FileSizeIsGreaterThanBufferSize_ReportsProperly ()
           StreamCopier.DefaultCopyBufferSize + 2000,
           2);
 
-      AssertBuildProgress (progressArgs[4],
+      AssertBuildProgress (
+          progressArgs[4],
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:601]: Unnecessary whitespace change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 13]
```diff
@@ -575,15 +597,17 @@ public void Build_FileSizeIsGreaterThanBufferSize_ReportsProperly ()
...
           2 * StreamCopier.DefaultCopyBufferSize + 1000 + 2000,
           StreamCopier.DefaultCopyBufferSize + 2000,
           1,
           @"root\file2",
           StreamCopier.DefaultCopyBufferSize + 2000,
           2);
 
-      AssertBuildProgress (progressArgs[5],
+      AssertBuildProgress (
+          progressArgs[5],
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:610]: Unnecessary whitespace change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 14]
```diff
@@ -675,46 +699,46 @@ public void Build_UsesProperFileShareToOpenTheFile (FileShare specifiedAdditionF
...
-          parent,
-          DateTime.Now,
-          DateTime.Now,
-          DateTime.Now);
+                 Path.Combine (parent.FullName, name),
+                 new NotClosableMemeoryStream (new byte[size]),
+                 parent,
+                 DateTime.Now,
+                 DateTime.Now,
+                 DateTime.Now);
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:707]: Unnecessary whitespace change

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 15]
```diff
@@ -675,46 +699,46 @@ public void Build_UsesProperFileShareToOpenTheFile (FileShare specifiedAdditionF
...
+                 else
+                 {
+                   var file = value as IFileInfo;
+                   if (file != null)
+                     directory.Files.Add (file);
+                 }
+               }
+
+               return directory;
+             };
```
Comment (Stefan-Ilic) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:741]: Unnecessary whitespace change

Comment (MichaelKetting) [reply Archive.Zip.UnitTests/ZipFileBuilderTest.cs:741]: How many of those whitespace-changes are due to the migration tool?

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 16]
```diff
@@ -190,16 +189,23 @@ public void NoHandlerForArchiveError_ThrowsException ()
...
-      }
-      finally
-      {
-        FileUtility.DeleteAndWaitForCompletion (zipFileName);
-      }
+
+      Assert.That (
+          () =>
+          {
+            try
```
Comment (MichaelKetting) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:196]: The try-finally should probably be outside the assert.that, just the using is inside the Assert.That.

File: Archive.Zip.UnitTests/ZipFileBuilderTest.cs [hunk 17]
```diff
@@ -255,16 +260,21 @@ public void SetFileProcessingRecoveryAction_Abort ()
...
-      }
-      finally
-      {
-        FileUtility.DeleteAndWaitForCompletion (zipFileName);
-      }
+      Assert.That (
+          () =>
+          {
+            try
+            {
```
Comment (MichaelKetting) [inline Archive.Zip.UnitTests/ZipFileBuilderTest.cs:267]: see above

**Review by Stefan-Ilic — APPROVED** [review]

**Review by MichaelKetting — APPROVED** [review]
